src/Microsoft.AspNet.Mvc.DataAnnotations/AttributeAdapterBase.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.ModelBinding.Validation;
+using Microsoft.Extensions.Localization;
+
+namespace Microsoft.AspNet.Mvc.DataAnnotations
+{
+    public interface IAttributeAdapter : IClientModelValidator
```

Comment:
Because we need a common class to get out of the Factory delegate and I don't know of any way to make it all both AttributeAdapterBase<MaxLengthAttributeAdapter> and AttributeAdapterBase<MinLengthAttributeAdapter> etc.